gnucash/gnome-utils/gnc-qrencode.c

```diff
@@ -0,0 +1,53 @@
+/********************************************************************\
+ * gnc-qrencode.c -- link to libqrencode                            *
+ * Copyright (C) 2021 Christopher Lam                               *
+ *                                                                  *
+ * This program is free software; you can redistribute it and/or    *
+ * modify it under the terms of the GNU General Public License as   *
+ * published by the Free Software Foundation; either version 2 of   *
+ * the License, or (at your option) any later version.              *
+ *                                                                  *
+ * This program is distributed in the hope that it will be useful,  *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of   *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the    *
+ * GNU General Public License for more details.                     *
+ *                                                                  *
+ * You should have received a copy of the GNU General Public License*
+ * along with this program; if not, contact:                        *
+ *                                                                  *
+ * Free Software Foundation           Voice:  +1-617-542-5942       *
+ * 51 Franklin Street, Fifth Floor    Fax:    +1-617-542-2652       *
+ * Boston, MA  02110-1301,  USA       gnu@gnu.org                   *
+\********************************************************************/
+
+#include <glib/gi18n.h>
+#include <qrencode.h>
+#include <stdio.h>
+
+#include "gnc-qrencode.h"
+
+SCM gnc_qrcode_encodestring (gchar *str)
+{
+    QRcode *qr = NULL;
+    SCM output = SCM_EOL;
+
+    qr = QRcode_encodeString (str, 0, QR_ECLEVEL_M, QR_MODE_8, FALSE);
+
+    if (!qr)
+    {
+        perror ("QRcode_encodeString() failed");
+        goto ret;
+    }
```

Comment:
@jralls is `perror` an acceptable error-dump mechanism here?

Reply:
No, it's [for printing strings based on system `ERRNO`](https://man7.org/linux/man-pages/man3/perror.3.html), plus it writes to `stderr` and that will just go to `/dev/null` unless the user happens to have started GnuCash from the command line.  Use `PERR`.